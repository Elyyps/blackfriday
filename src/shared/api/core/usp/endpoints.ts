import { uniqueSellingPoints } from "./generate-dummy-data";
import { USP } from "./usp";

export const generateUSP = (): USP => uniqueSellingPoints;
