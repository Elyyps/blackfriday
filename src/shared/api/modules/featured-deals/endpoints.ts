import { dummyFeaturedDealsData } from "./generate-dummy-data";

export const getFeaturedDealsDummyData = () => dummyFeaturedDealsData;
