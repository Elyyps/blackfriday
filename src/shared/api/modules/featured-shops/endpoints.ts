import { dummyFeaturedShopsData } from "./generate-dummy-data";

export const getFeaturedShopsDummyData = () => dummyFeaturedShopsData;
