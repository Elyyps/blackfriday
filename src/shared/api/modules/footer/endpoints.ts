import { bottomFooter, middleFooter, topFooter } from "./generate-dummy-data";
import { Footer } from "./models/footer.module";

export const getFullfooter = (): Footer => ({
  bottomFooter: bottomFooter(),
  middleFooter: middleFooter(),
  topFooter: topFooter()
});
