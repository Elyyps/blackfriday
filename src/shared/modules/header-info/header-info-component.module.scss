.header-info {
  position: relative;

  &__image {
    height: 424px;
    overflow: hidden;
    position: relative;
    margin-bottom: -80px;
    z-index: 1;
    @media (max-width: $breakpoint-small-max) {
      height: 250px;
      margin: 0;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      height: 216px;
    }
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
  &__body {
    position: relative;
    z-index: 2;
    display: flex;
    align-items: center;
    background: $global-background;
    border-radius: 2px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
    padding: 40px 194px 40px 40px;
    color: $gray-light;
    &__content {
      display: flex;
    }
    @media (max-width: $breakpoint-small-max) {
      padding: 52px 32px 32px;
      margin: 0 -24px;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 52px 24px 24px;
      margin: 0 -16px;
    }
    a {
      color: $gray-dark;
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  }

  &__share-dropdown {
    position: absolute;
    right: 60px;
    top: 50px;
    z-index: 2;
    @media (max-width: $breakpoint-small-max) {
      display: none;
    }
  }

  &__share-badge {
    display: none;
    @media (max-width: $breakpoint-small-max) {
      display: block;
    }
  }
  &__icon {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-shrink: 0;
    width: 68px;
    height: 68px;
    border-radius: 50%;
    border: solid 2px $tint-gray-light;
    margin-right: 28px;
    background: $global-background;
    @media (max-width: $breakpoint-small-max) {
      position: absolute;
      top: -36px;
      left: 24px;
    }
  }
  &__icon-image {
    width: 50%;
    height: 50%;
  }
  &__title {
    margin-bottom: 10px;
    font-family: $global-font-family-secondary;
    font-size: 32px;
    color: $gray-dark;
  }
  &__content {
    flex: 1 1 auto;
  }
}

.breadcrumbs {
  display: flex;
  align-items: flex-start;
}
