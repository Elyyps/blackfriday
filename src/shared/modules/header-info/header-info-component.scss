.header-info {
  position: relative;

  &__image {
    height: 424px;
    overflow: hidden;
    position: relative;
    margin-bottom: -80px;
    @media (max-width: $breakpoint-small-max) {
      height: 250px;
      margin: 0;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      height: 216px;
    }
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
  &__body {
    position: relative;
    z-index: 2;
    display: flex;
    align-items: center;
    background: $global-background;
    border-radius: 2px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
    padding: 40px 194px 40px 40px;
    color: $gray-light;
    @media (max-width: $breakpoint-small-max) {
      padding: 52px 32px 32px;
      margin: 0 -24px;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 52px 24px 24px;
      margin: 0 -16px;
    }
    a {
      color: $gray-dark;
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  }
  &__icon {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-shrink: 0;
    width: 68px;
    height: 68px;
    border-radius: 50%;
    border: solid 2px $tint-gray-light;
    margin-right: 28px;
    background: $global-background;
    @media (max-width: $breakpoint-small-max) {
      position: absolute;
      top: -36px;
      left: 24px;
    }

    img {
      width: 35px;
      height: 35px;
      object-fit: cover;
    }
  }
  &__title {
    margin-bottom: 10px;
  }
  &__content {
    flex: 1 1 auto;
  }

  .shared-box {
    position: absolute;
    right: 40px;
    top: 40px;
    @media (max-width: $breakpoint-small-max) {
      right: 0;
      top: 0;
      .button--dropdown {
        line-height: 0;
        font-size: 0;
        width: 56px;
        height: 56px;
        padding: 0 0 0 13px;
        background: none;
        overflow: hidden;
        position: relative;
        &:hover,
        &.uk-open {
          box-shadow: none;
        }
        &::before {
          width: 100px;
          height: 100px;
          background: $tint-gray-light;
          position: absolute;
          border-radius: 100%;
          left: 0;
          bottom: 0;
          content: "";
          z-index: -1;
        }
        .button__arrow {
          display: none;
        }
      }
    }
  }
}
