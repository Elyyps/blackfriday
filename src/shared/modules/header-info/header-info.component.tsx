import * as React from "react";
import styles from "./header-info-component.module.scss";
import { ImageComponent } from "@app/core/image";
import { DropdownComponent } from "@app/core/dropdown/dropdown.component";
import { ShareSocialComponent } from "@app/core/share-social";
import { HeaderInfo } from "@app/api/modules/header-info/header-info.module";
import { generateShareSocialModule } from "@app/api/core/share-social/endpoints";
import { Badge } from "@app/core/badge";

import ShareSVG from "@assets/icons/share.svg";

import { LinkComponent } from "@app/core/link";
import { BreadcrumbComponent } from "@app/core/breadcrumb";

export interface IHeaderInfoComponentProps {
  headerInfo: HeaderInfo;
}

const HeaderInfoComponent = (props: IHeaderInfoComponentProps) => (
  <div className={styles["header-info"]} style={{ background: props.headerInfo.bgcolor }}>
    {props.headerInfo.image && (
      <div className={styles["header-info__image"]}>
        <ImageComponent src={props.headerInfo.image} />
      </div>
    )}
    <div className="uk-container">
      <div className={styles["header-info__body"]}>
        <div className={styles["header-info__body__content"]}>
          {props.headerInfo.icon && (
            <div className={styles["header-info__icon"]}>
              <ImageComponent src={props.headerInfo.icon} />
            </div>
          )}
          <div className={styles["header-info__content"]}>
            {props.headerInfo.breadcrumbProps && (
              <div className={styles["breadcrumbs"]}>
                <BreadcrumbComponent
                  links={props.headerInfo.breadcrumbProps.links}
                  backButton={props.headerInfo.breadcrumbProps.backButton}
                  variant={props.headerInfo.breadcrumbProps.variant}
                />
              </div>
            )}
            <h1 className={styles["header-info__title"]}>{props.headerInfo.title}</h1>
            {`${props.headerInfo.content} `}
            <LinkComponent to={props.headerInfo.url}>Meer info</LinkComponent>
          </div>
          <div className={styles["header-info__share-dropdown"]}>
            <DropdownComponent title="Status" icon={ShareSVG}>
              <ShareSocialComponent {...generateShareSocialModule()} />
            </DropdownComponent>
          </div>
        </div>
        <div className={styles["header-info__share-badge"]}>
          <Badge icon={ShareSVG} position={"right"}>
            <ShareSocialComponent {...generateShareSocialModule()} />
          </Badge>
        </div>
      </div>
    </div>
  </div>
);

export { HeaderInfoComponent };
