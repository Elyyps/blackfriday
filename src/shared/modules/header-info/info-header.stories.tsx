import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";

import { HeaderInfoComponent } from "../header-info";

import { getHeaderInfoDummyData } from "@app/api/modules/header-info/endpoints";
import { withProvider } from "@app/util";

storiesOf("InfoHeader", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <HeaderInfoComponent headerInfo={getHeaderInfoDummyData()} />);
