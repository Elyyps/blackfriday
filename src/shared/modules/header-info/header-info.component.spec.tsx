import React from "react";
import { shallow } from "enzyme";
import { HeaderInfoComponent } from "../header-info";

import { getHeaderInfoDummyData } from "@app/api/modules/header-info/endpoints";

describe("[Footer]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HeaderInfoComponent headerInfo={getHeaderInfoDummyData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
