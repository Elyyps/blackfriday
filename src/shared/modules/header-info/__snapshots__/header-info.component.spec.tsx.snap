// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Footer] should render component without crashing 1`] = `
<div
  className="header-info"
  style={
    Object {
      "background": "#f7f7f7 ",
    }
  }
>
  <div
    className="header-info__image"
  >
    <ImageComponent
      src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/apple-iphones-header.jpg"
    />
  </div>
  <div
    className="uk-container"
  >
    <div
      className="header-info__body"
    >
      <div
        className="header-info__body__content"
      >
        <div
          className="header-info__icon"
        >
          <ImageComponent
            src="store.svg"
          />
        </div>
        <div
          className="header-info__content"
        >
          <div
            className="breadcrumbs"
          >
            <BreadcrumbComponent
              backButton={
                Object {
                  "title": "Terug",
                  "url": "/#",
                }
              }
              links={
                Array [
                  Object {
                    "title": "Home",
                    "url": "/",
                  },
                  Object {
                    "title": "Alle winkels",
                    "url": "/stores-overview",
                  },
                ]
              }
              variant="dark"
            />
          </div>
          <h1
            className="header-info__title"
          >
            Black Friday 2019 bij Apple
          </h1>
          Vier Black Friday bij Apple en ontvang 15% korting op alles! 
          <LinkComponent
            to="./"
          >
            Meer info
          </LinkComponent>
        </div>
        <div
          className="header-info__share-dropdown"
        >
          <DropdownComponent
            icon="share.svg"
            title="Status"
          >
            <ShareSocialComponent
              email={
                Object {
                  "nameSocialShare": "Email",
                  "subject": "BlackFriday",
                  "title": "Email",
                  "url": "https://www.webbio.nl",
                }
              }
              facebook={
                Object {
                  "nameSocialShare": "Facebook",
                  "title": "BlackFriday",
                  "url": "https://www.webbio.nl",
                }
              }
              twitter={
                Object {
                  "hashtags": Array [
                    "BlackFriday",
                    "Netherlands",
                    "Sale",
                    "Shops",
                  ],
                  "nameSocialShare": "Twitter",
                  "title": "BlackFriday",
                  "url": "https://www.webbio.nl",
                  "via": "BlackFriday",
                }
              }
              whatsapp={
                Object {
                  "nameSocialShare": "WhatsApp",
                  "separator": " - ",
                  "title": "BlackFriday",
                  "url": "https://www.webbio.nl",
                }
              }
            />
          </DropdownComponent>
        </div>
      </div>
      <div
        className="header-info__share-badge"
      >
        <Badge
          icon="share.svg"
          position="right"
        >
          <ShareSocialComponent
            email={
              Object {
                "nameSocialShare": "Email",
                "subject": "BlackFriday",
                "title": "Email",
                "url": "https://www.webbio.nl",
              }
            }
            facebook={
              Object {
                "nameSocialShare": "Facebook",
                "title": "BlackFriday",
                "url": "https://www.webbio.nl",
              }
            }
            twitter={
              Object {
                "hashtags": Array [
                  "BlackFriday",
                  "Netherlands",
                  "Sale",
                  "Shops",
                ],
                "nameSocialShare": "Twitter",
                "title": "BlackFriday",
                "url": "https://www.webbio.nl",
                "via": "BlackFriday",
              }
            }
            whatsapp={
              Object {
                "nameSocialShare": "WhatsApp",
                "separator": " - ",
                "title": "BlackFriday",
                "url": "https://www.webbio.nl",
              }
            }
          />
        </Badge>
      </div>
    </div>
  </div>
</div>
`;
